docs/spec.md

```diff
@@ -0,0 +1,288 @@
+
+#### EX01 - Completion for `Condition` attributes
+
+- **Requirement**: *When the user types inside a conditional attribute the MSBuild Language Service shall display completions for defined items that are in scope.*
+- **Additional Details**: See documentation [here](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-conditions?view=vs-2019)
+
+#### EX02 - Completion for `$()` properties
+
+- **Requirement**: *When the user types inside a `$()` expression the MSBuild Language Service shall display completions for defined properties that are in scope.*
+- **Additional Details**: See documentation [here](https://docs.microsoft.com/en-us/visualstudio/msbuild/special-characters-to-escape?view=vs-2019)
+
+#### EX03 - Completion for `@()` items
+
+- **Requirement**: *When the user types inside a `%()` expression the MSBuild Language Service shall display completions for defined items that are in scope.*
+
+#### EX04  Completion for `%()` item metadata
+
+##### EX04.1 - well-known item metadata
+
+- **Requirement**: *When the user types inside a `%()` expression the MSBuild Language Service shall display completions for well-known item metadata.*
+- **Additional Details**: See documentation [here](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-well-known-item-metadata?view=vs-2019)
+
+##### EX04.2 - user defined item metadata
+
+- **Requirement**: *When the user types inside a `%()` expression the MSBuild Language Service shall display completions for defined metadata that are in scope.*
+- **Additional Details**: See documentation [here](https://docs.microsoft.com/en-us/visualstudio/msbuild/special-characters-to-escape?view=vs-2019)
+
+#### EX05 - Completion for task metadata
+
+- **Requirement**: *When the user types inside a `%()` expression the MSBuild Language Service shall display completions for task metadata that are in scope.*
+- **Additional Details**: See documentation [here](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-tasks?view=vs-2019)
+
+#### WRN1 - Fade out unnecessary properties
+
+- **Requirement**: *When the user opens a files that contains properties that are re-defined and re-assigned to values they already have the MSBuild Language Service shall indicate that these properties are unnecessary.*
+
+#### WRN2 - Warn if property doesn't exist
+
+- **Requirement**: *When the user opens a files that contains expressions which refer to non-existant properties the MSBuild Language Service shall indicate that these properties are not defined.*
+
+## Design Decisions
+
+### Technology Decisions
+
+The language-server-protocol is currently the best cross-platform technology we have for building a language service that works in multiple IDEs. It therefore seems like the correct initial approach is to start with a lsp implementation initially.

```

Comment:
Have we verified that LSP can support all the features we want to implement?

Reply:
The protocol is extensible. So we should not be limited by anything